server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/node/FixedLengthStarTreeNode.java

```diff
@@ -0,0 +1,276 @@
+        this.in = in;
+        this.nodeId = nodeId;
+        firstChildId = getInt(FIRST_CHILD_ID_OFFSET);
+    }
+
+    /**
+     * Reads an integer value from the specified offset in the node's data.
+     *
+     * @param fieldOffset The offset of the field to read
+     * @return The integer value at the specified offset
+     * @throws IOException If there's an error reading from the input
+     */
+    private int getInt(int fieldOffset) throws IOException {
+        return in.readInt(nodeId * SERIALIZABLE_DATA_SIZE_IN_BYTES + fieldOffset);
+    }
+
+    /**
+     * Reads a long value from the specified offset in the node's data.
+     *
+     * @param fieldOffset The offset of the field to read
+     * @return The long value at the specified offset
+     * @throws IOException If there's an error reading from the input
+     */
+    private long getLong(int fieldOffset) throws IOException {
+        return in.readLong(nodeId * SERIALIZABLE_DATA_SIZE_IN_BYTES + fieldOffset);
+    }
+
+    /**
+     * Reads a byte value from the specified offset in the node's data.
+     *
+     * @param fieldOffset The offset of the field to read
+     * @return The byte value at the specified offset
+     * @throws IOException If there's an error reading from the input
+     */
+    private byte getByte(int fieldOffset) throws IOException {
+        return in.readByte(nodeId * SERIALIZABLE_DATA_SIZE_IN_BYTES + fieldOffset);
+    }
+
+    @Override
+    public int getDimensionId() throws IOException {
+        return getInt(DIMENSION_ID_OFFSET);
+    }
+
+    @Override
+    public long getDimensionValue() throws IOException {
+        return getLong(DIMENSION_VALUE_OFFSET);
+    }
+
+    @Override
+    public int getChildDimensionId() throws IOException {
+        if (firstChildId == INVALID_ID) {
+            return INVALID_ID;
+        } else {
+            return in.readInt(firstChildId * SERIALIZABLE_DATA_SIZE_IN_BYTES);
+        }
+    }
+
+    @Override
+    public int getStartDocId() throws IOException {
+        return getInt(START_DOC_ID_OFFSET);
+    }
+
+    @Override
+    public int getEndDocId() throws IOException {
+        return getInt(END_DOC_ID_OFFSET);
+    }
+
+    @Override
+    public int getAggregatedDocId() throws IOException {
+        return getInt(AGGREGATE_DOC_ID_OFFSET);
+    }
+
+    @Override
+    public int getNumChildren() throws IOException {
+        if (firstChildId == INVALID_ID) {
+            return 0;
+        } else {
+            return getInt(LAST_CHILD_ID_OFFSET) - firstChildId + 1;
+        }
+    }
+
+    @Override
+    public boolean isLeaf() {
+        return firstChildId == INVALID_ID;
+    }
+
+    @Override
+    public byte getStarTreeNodeType() throws IOException {
+        return getByte(STAR_NODE_TYPE_OFFSET);
+    }
+
+    @Override
+    public StarTreeNode getChildForDimensionValue(Long dimensionValue, boolean isStar) throws IOException {
```

Comment:
Its difficult to understand how this will be used. How will caller know whether it is reading a star node or not?

Reply:
I can split the ```getChildForDimensionValue``` into two. One would only be to query star node and rest of the child nodes can be iterated via this function.

If the dimension to search is not specified, then it can tell us that its a star node search.